### PR TITLE
docs: add jsonschema

### LINKED ISSUE #1453: config file json schema
**Is your feature request related to a problem? Please describe.**

I have written a JSON Schema for golangci-lint's config file. Maybe someone can check or even improve it? Some of the descriptions could be improved upon also :wink:
https://github.com/SchemaStore/schemastore/pull/1312

This will enable autocompletion, tooltips and validation in IDEs like VSCode or JetBrains.

**Describe the solution you'd like**

A second (or even more) pair of eyes to check the schema. Maybe you have some example config files that the schema can be tested against.

I can also add a PR in this repository to add the schema. In other projects I already utilize documentation generation of pages like your https://golangci-lint.run/usage/configuration/#config-file page from the schema.

**Describe alternatives you've considered**

Just add it as is to the schemastore.

**Additional context**

One of the auto generated docs pages from one of my projects: https://www.ory.sh/kratos/docs/reference/configuration

### LINKED ISSUE #4486: Update the JSON schema
### Your feature request related to a problem? Please describe.

#4437 adds a new kind of plugin. The [published JSON schema](https://json.schemastore.org/golangci-lint.json) needs to be updated. Specifically the current schema does not allow `linters-settings.custom.{name}.path` to be unset. It also doesn't have any of the new properties, but I don't care about that as much as the angry red squiggles from the supposedly missing property.

### Describe the solution you'd like.

The schema to be updated.

### Describe alternatives you've considered.

N/A

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by ldez**
- copy the JSON schema from https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/golangci-lint.json
- add the next JSON schema (v1.57.0)

The schema will be published here:
- https://raw.githubusercontent.com/golangci/golangci-lint/master/jsonschema/golangci.jsonschema.json
- https://raw.githubusercontent.com/golangci/golangci-lint/master/jsonschema/golangci.next.jsonschema.json
- https://raw.githubusercontent.com/golangci/golangci-lint/master/jsonschema/custom-gcl.next.jsonschema.json

When this PR is merged, I will create a PR on the schema store to reference our schema.

Fixes #4486
Fixes #1453

### CODE REVIEW

File: .github/workflows/post-release.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ jobs:
         run: cp .golangci.next.reference.yml .golangci.reference.yml
 
       - name: Update JSON schema files
-        run: cp docs/static/schema/next.jsonschema.json docs/static/schema/jsonschema.json
+        run: cp .schema/next.jsonschema.json .schema/jsonschema.json
```
Comment (bombsimon) [inline .github/workflows/post-release.yml:69]: If this is current workdir, I think `./` is more common and less confusing unless it's a hidden folder in cwd.

```suggestion
        run: cp ./schema/next.jsonschema.json ./schema/jsonschema.json
```

**Comment by ldez** [comment]
Sorry, I moved the schema after your review: the files were too deep in the tree, and I think will forget them.
I think we don't need to publish them on the domain `golangci-lint.run`.

File: .github/workflows/post-release.yml (hunk 1 above)
Comment (ldez) [reply .github/workflows/post-release.yml]: I changed to `jsonschema`

**Comment by ldez** [comment]
I renamed the file to follow the same name as the config file names.

**Pull request merged**